test/modules/kpi-reporting/controller.js

```diff
@@ -0,0 +1,71 @@
+'use-strict';
+const { experiment, test, beforeEach, afterEach } = exports.lab = require('@hapi/lab').script();
+const { expect } = require('@hapi/code');
+const sandbox = require('sinon').createSandbox();
+const controller = require('../../../src/modules/kpi-reporting/controller');
+const repos = require('../../../src/lib/repo/kpi-reporting');
+
+experiment('./modules/kpi-reporting/controller', async () => {
+  const repoData = {
+    data: [
+      {
+        current_year: true,
+        total: 10,
+        month: 1,
+        year: 2020
+      },
+      {
+        current_year: true,
+        total: 15,
+        month: 2,
+        year: 2020
+      },
+      {
+        current_year: true,
+        total: 5,
+        month: 3,
+        year: 2020
+      },
+      {
+        current_year: false,
+        total: 100,
+        month: 12,
+        year: 2019
+      }
+    ],
+    error: null
+  };
+
+  afterEach(async => {
+    sandbox.restore();
+  });
+
+  experiment('when data is returned from the repo', async () => {
+    beforeEach(async => {
+      sandbox.stub(repos, 'getEntityRolesKPIdata').resolves(repoData);
+    });
+    test('the right data shape is returned', async () => {
+      const { data } = await controller.getKPIEntityRolesData();
+      expect(data.totals.allTime).to.equal(130);
+      expect(data.totals.ytd).to.equal(30);
+      expect(data.monthly).to.be.array();
+      expect(Object.keys(data.monthly[0])).to.equal(['month', 'total', 'change']);
```

Comment:
I'm not sure that these are required as they are confirmed by the 3 tests below

Reply:
Thanks - removed that line.